### PR TITLE
[MINOR][DOC] Update the building doc to use Maven 3.5.4 and Java 8 only

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
## What changes were proposed in this pull request?

Since we didn't test Java 9 ~ 11 up to now in the community, fix the document to describe Java 8 only.

## How was this patch tested?
N/A (This is a document only change.)

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

File: docs/building-spark.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ redirect_from: "building-with-maven.html"
 ## Apache Maven
 
 The Maven-based build is the build of reference for Apache Spark.
-Building Spark using Maven requires Maven 3.3.9 or newer and Java 8+.
+Building Spark using Maven requires Maven 3.3.9 or newer and Java 8.
```
Comment (dongjoon-hyun) [inline docs/building-spark.md]: Ur, @kiszk and @cloud-fan , @HyukjinKwon .

It seems to be Maven version is also outdated. We updated Maven version in  https://github.com/apache/spark/pull/21905 and https://github.com/apache/spark/pull/21920 .

Should we change it together? Or, Maven 3.3.9 is still valid to compile Spark?

Comment (cloud-fan) [reply docs/building-spark.md]: let's change to the expected maven version, thanks!

Comment (HyukjinKwon) [reply docs/building-spark.md]: +1

Comment (dongjoon-hyun) [reply docs/building-spark.md]: Thank you, @cloud-fan and @HyukjinKwon .

File: docs/building-spark.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ redirect_from: "building-with-maven.html"
 ## Apache Maven
 
 The Maven-based build is the build of reference for Apache Spark.
-Building Spark using Maven requires Maven 3.3.9 or newer and Java 8+.
+Building Spark using Maven requires Maven 3.5.4 and Java 8.
```
Comment (dongjoon-hyun) [inline docs/building-spark.md]: Maven 3.5.4 is the latest one and there is no newer version for now. We cannot guarantee the newer Maven work or not.

**Review by felixcheung — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you for review, @HyukjinKwon , @cloud-fan , @felixcheung .

**Comment by dongjoon-hyun** [comment]
Merged to master/branch-2.4.

**Pull request closed without merging**